core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java

```diff
@@ -209,7 +211,16 @@ public void abort() {
     @Override
     public boolean write(T obj) {
         requireNonNull(obj, "obj");
+        if (obj instanceof ReferenceCounted) {
```

Comment:
We still have to check if `obj` is a `ReferenceCounted` because otherwise it will raise an `IllegalArgumentException` for the object that's neither a `ByteBufHolder` nor a `ByteBuf`. e.g. `DefaultHttpData`.